core/src/main/java/org/apache/iceberg/io/SortedPosDeleteWriter.java

```diff
@@ -59,20 +60,58 @@ class SortedPosDeleteWriter<T> implements FileWriter<PositionDelete<T>, DeleteWr
       OutputFileFactory fileFactory,
       FileFormat format,
       StructLike partition,
-      long recordsNumThreshold) {
+      long recordsNumThreshold,
+      DeleteGranularity deleteGranularity) {
     this.appenderFactory = appenderFactory;
     this.fileFactory = fileFactory;
     this.format = format;
     this.partition = partition;
     this.recordsNumThreshold = recordsNumThreshold;
+    this.deleteGranularity = deleteGranularity;
+  }
+
+  SortedPosDeleteWriter(
```

Comment:
These constructors are not utilised by our code (or only by tests), but kept them for compatibility reasons.
Do we need them? Are they part of the API?

Reply:
reverted